lib/ansible/modules/cloud/amazon/aws_glue_job.py

```diff
@@ -231,9 +231,9 @@ def _compare_glue_job_params(user_params, current_params):
         return True
     if 'Command' in user_params and user_params['Command']['ScriptLocation'] != current_params['Command']['ScriptLocation']:
         return True
-    if 'Connections' in user_params and set(user_params['Connections']) != set(current_params['Connections']):
+    if 'Connections' in user_params and user_params['Connections'] != current_params['Connections']:
```

Comment:
This comparison should work well for DefaultArguments since the keys and values are strings, but I'm not sure about Connections since it looks like `{'Connections': {'Connections': [...,]}}`. Does order have significance here? If not, the list should be sorted before comparison (or the list should be cast to a set rather than the dict if AWS removes duplicates from the list). It would be good to have tests for this module. If you have some playbooks, they would be a good starting place. It looks like we have tests for[ aws_glue_connection](https://github.com/ansible/ansible/tree/devel/test/integration/targets/aws_glue_connection) to see an example.